src/main/kotlin/org/onflow/flow/sdk/crypto/Crypto.kt

```diff
@@ -156,11 +159,31 @@ object Crypto {
 }
 
 internal class HasherImpl(
-    private val hashAlgo: HashAlgorithm
+    private val hashAlgo: HashAlgorithm,
+    private val key: ByteArray? = null
 ) : Hasher {
     override fun hash(bytes: ByteArray): ByteArray {
-        val digest = MessageDigest.getInstance(hashAlgo.algorithm)
-        return digest.digest(bytes)
+        return when (hashAlgo) {
+            HashAlgorithm.KECCAK256 -> {
+                val keccakDigest = Keccak.Digest256()
+                keccakDigest.digest(bytes)
+            }
+            HashAlgorithm.KMAC128 -> {
+                if (key == null) {
+                    throw IllegalArgumentException("KMAC128 requires a key")
+                }
+                val kmac = KMAC(128, ByteArray(0))
+                kmac.init(KeyParameter(key))
+                val output = ByteArray(kmac.digestSize)
+                kmac.update(bytes, 0, bytes.size)
```

Comment:
I think we could simply call 
```suggestion
                kmac.update(bytes)
```